src/components/Cardlist/cardlist.tsx

```diff
@@ -0,0 +1,72 @@
+import { useEffect, useRef, useState } from "react";
+import Cards from "../Cards/cards";
+import { ArticlesData, articles } from "../../main";
+import "./cardlist.css";
+import "./search.css";
+
+const Cardlist = (): JSX.Element => {
```

Comment:
- [ ] Rename this Component to something like **ArticlesModule**
- [ ] you can refactor all the search logic and template to an extra component